File: src/main/java/com/atsuishio/superbwarfare/entity/vehicle/LaserTowerEntity.java
package com.atsuishio.superbwarfare.entity.vehicle;

import com.atsuishio.superbwarfare.entity.vehicle.base.AutoAimableEntity;
import com.atsuishio.superbwarfare.init.ModEntities;
import net.minecraft.world.entity.EntityType;
import net.minecraft.world.level.Level;
import net.minecraftforge.network.PlayMessages;

public class LaserTowerEntity extends AutoAimableEntity {

    public LaserTowerEntity(PlayMessages.SpawnEntity packet, Level world) {
        this(ModEntities.LASER_TOWER.get(), world);
    }

    public LaserTowerEntity(EntityType<LaserTowerEntity> type, Level world) {
        super(type, world);
        this.noCulling = true;
    }
}

File: src/main/kotlin/com/atsuishio/superbwarfare/entity/vehicle/LaserTowerEntity.kt
package com.atsuishio.superbwarfare.entity.vehicle

import com.atsuishio.superbwarfare.entity.vehicle.base.AutoAimableEntity
import net.minecraft.world.entity.EntityType
import net.minecraft.world.level.Level

open class LaserTowerEntity(type: EntityType<LaserTowerEntity>, world: Level) : AutoAimableEntity(type, world), BasicGeoVehicleEntity {
    init {
        this.noCulling = true
    }
}
